bin/mediatis-typo3-coding-standards-setup

```diff
@@ -1,30 +1,81 @@
 #!/usr/bin/env php
 <?php
 
-$supportedTypo3Versions = [11, 12];
+use Composer\Semver\Semver;
+use Composer\Semver\VersionParser;
+use Symfony\Component\Yaml\Yaml;
+use TYPO3\CMS\Core\Utility\ArrayUtility;
+
+function includeIfExists(string $file): bool
+{
+    return file_exists($file) && include $file;
+}
+
+if (
+    !includeIfExists(__DIR__ . '/../../../autoload.php')
+    && !includeIfExists(__DIR__ . '/../vendor/autoload.php')
+    && !includeIfExists(__DIR__ . '/../../../../../vendor/autoload.php')
+) {
+    fwrite(STDERR, 'Install dependencies using Composer.' . PHP_EOL);
+    exit(1);
+}
+
+$supportedTypo3PackageKeys = ['typo3/cms', 'typo3/cms-core'];
+$supportedTypo3Versions = [11.5, 12.4];
+$supportedPhpVersions = [8.1, 8.2];
+$supportedPhpPackageKeys = ['php'];
 $binDirectory = $_composer_bin_dir ?? __DIR__ . '/../../../../bin';
 $targetPackageDirectory = realpath($binDirectory . '/../..');
 $codingStandardsPackageDirectory = realpath(__DIR__ . '/..');
 $requiredFolderPaths = ['Classes', 'Tests'];
```

Comment:
@mkarulin, I tried out the package with `digital-marketing-framework/typo3-core` starting from `main` to get a fresh state.

When I executed `composer ci:tests` I got the script complaining about the folder `Tests/Unit` not existing.

I changed the line...
```
$requiredFolderPaths = ['Classes', 'Tests'];
```
... to...
```
$requiredFolderPaths = ['Classes', 'Tests/Unit', 'Tests/Functional'];
```

But now I get this error:
```
> find 'Tests/Functional' -wholename '*Test.php' | parallel --gnu 'echo; echo "Running functional test suite {}"; .Build/bin/phpunit -c .Build/vendor/typo3/testing-framework/Resources/Core/Build/FunctionalTests.xml {}';
sh: parallel: command not found

Script find 'Tests/Functional' -wholename '*Test.php' | parallel --gnu 'echo; echo "Running functional test suite {}"; .Build/bin/phpunit -c .Build/vendor/typo3/testing-framework/Resources/Core/Build/FunctionalTests.xml {}'; handling the ci:tests:functional event returned with error code 127
Script @ci:tests:functional was called via ci:tests
```

Does it work on gitlab and github? Is the `parallel` command something that should be available on development machines? Unfortunately, we cannot assume a hosting project present, so using `ddev` to make `parallel` available is (currently) not an option.